Source/WebCore/platform/graphics/NativeImage.cpp

```diff
@@ -104,11 +104,45 @@ DestinationColorSpace NativeImage::colorSpace() const
     return m_backend->colorSpace();
 }
 
+bool NativeImage::hasHDRContent() const
+{
+    return colorSpace().usesITUR_2100TF();
+}
+
 Headroom NativeImage::headroom() const
 {
     return m_backend->headroom();
 }
 
+void NativeImage::drawToneMapping(GraphicsContext& context, const FloatRect& destinationRect, const FloatRect& sourceRect, ImagePaintingOptions options)
+{
+    auto colorSpaceForToneMapping = [](GraphicsContext& context) {
+#if PLATFORM(IOS_FAMILY)
+        // iOS typically renders into extended range sRGB to preserve wide gamut colors, but we want
```

Comment:
This comment seems to be conflating "extended range" with "extended dynamic range". P3 is an extended range colorspace (values outside of sRGB).

Reply:
I fixed the comment.